neural_compressor/onnxrt/algorithms/smooth_quant/algo_entry.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+from pathlib import Path
+from typing import Union
+
+import onnx
+from onnxruntime.quantization import StaticQuantConfig, quantize
+
+from neural_compressor.common import Logger
+from neural_compressor.common.utils import SMOOTH_QUANT
+from neural_compressor.onnxrt.algorithms.smooth_quant.smooth_quant import ORTSmoothQuant
+from neural_compressor.onnxrt.quantization import CalibrationDataReader
+from neural_compressor.onnxrt.quantization.config import SmoohQuantQuantConfig
+from neural_compressor.onnxrt.utils.utility import register_algo
+
+logger = Logger().get_logger()
+
+
+@register_algo(name=SMOOTH_QUANT)
```

Comment:
@mengniwang95  I would suggest to align with torch algorithms layout to unify the code structure. 

../torch/algorithms/weight_only_algos.py:29:@register_algo(name=RTN)
../torch/algorithms/weight_only_algos.py:47:@register_algo(name=GPTQ)

Reply:
<img width="221" alt="image" src="https://github.com/intel/neural-compressor/assets/65003808/35819f1b-6dc9-4016-8ced-18e65bfa4520">
torch removes this file in another PR, it seems the code structure is not stable yet.

Reply:
Yes, we prefer to keep algorithm independent so the registration will happen in `quantization` folder.

Reply:
[refactor folder of torch 3x API and refine RTN by xin3he · Pull Request #1560 · intel/neural-compressor (github.com)](https://github.com/intel/neural-compressor/pull/1560)

Reply:
It would be better if you can let us know as soon as there is a stable structure so that we can save more time on develop.